src/EntityComponentManager.cc

```diff
@@ -945,15 +945,6 @@ void EntityComponentManager::AddEntityToMessage(msgs::SerializedStateMap &_msg,
   // Add a component to the message and set it to be removed if the component
   // exists in the removedComponents map.
   this->dataPtr->SetRemovedComponentsMsgs(_entity, _msg);
-
-  // Remove the entity from the message if a component for the entity was
-  // not modified or added. This will allow the state message to shrink.
-  if (entIter == _msg.mutable_entities()->end() &&
-      (entIter = _msg.mutable_entities()->find(_entity)) !=
-      _msg.mutable_entities()->end())
-  {
-    _msg.mutable_entities()->erase(entIter);
```

Comment:
The ABI checker gave what I assumed was a false positive, see https://github.com/ignition-tooling/release-tools/issues/336

I inspected the symbols locally just in case, and was surprised to see that this symbol was actually gone:

`_ZNK8ignition6gazebo2v410components13BaseComponent9SerializeERSo`

I narrowed it down to line 955. When it's removed, the symbol disappears with it :fearful: :woman_shrugging: 

You can see all symbols before and after removing the line here: https://gist.github.com/chapulina/7f53a3d43cb031715f005650e582fcb7

Reply:
Looks like that symbol is not used by anything in `libignition-gazebo.so` so it must be getting optimized away. If you build in debug mode, it doesn't get removed.

Since the function is defined in a header, I think this is somewhat of a false positive. Any translation unit that includes the `Component.hh` header will have access to symbol. For example,  `UNIT_Component_TEST` still has the symbol even though this it is missing from `libignition-gazebo.so`.

Reply:
Hum interesting and mysterious how the compiler decides to remove the symbol. Your reasoning about the ABI not being broken makes sense to me. I guess all functions defined in headers may be subject to the same rule.

Makes me wonder about why the symbol would be added in the first place. When the compiler does add the symbol, I guess it can be used directly instead of regenerated when the header is included?

This messes up the poor ABI checker :confused: